src/metal.cs

```diff
@@ -1386,9 +1683,75 @@ public interface MTLRenderPassDescriptor : NSCopying {
 		[Static, Export ("renderPassDescriptor")]
 		[Autorelease]
 		MTLRenderPassDescriptor CreateRenderPassDescriptor ();
+		
+		[NoiOS, Mac (10,11)]
+		[Export ("renderTargetArrayLength")]
+		nuint RenderTargetArrayLength { get; set; }
 	}
 
+
+	[iOS (10, 0), TV (10,0), NoWatch, NoMac]
+	[BaseType (typeof(NSObject))]
+	public interface MTLHeapDescriptor : NSCopying
+	{
+		[Export ("size")]
+		nuint Size { get; set; }
+
+		[Export ("storageMode", ArgumentSemantic.Assign)]
+		MTLStorageMode StorageMode { get; set; }
+
+		[Export ("cpuCacheMode", ArgumentSemantic.Assign)]
+		MTLCpuCacheMode CpuCacheMode { get; set; }
+	}
+	
+	[iOS (10, 0), TV (10,0), NoWatch, NoMac]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	public interface MTLHeap
+	{
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+
+		[Abstract]
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[Export ("storageMode")]
+		MTLStorageMode StorageMode { get; }
+
+		[Abstract]
+		[Export ("cpuCacheMode")]
+		MTLCpuCacheMode CpuCacheMode { get; }
+
+		[Abstract]
+		[Export ("size")]
+		nuint Size { get; }
+
+		[Abstract]
+		[Export ("usedSize")]
+		nuint UsedSize { get; }
+
+		[Abstract]
+		[Export ("maxAvailableSizeWithAlignment:")]
+		nuint MaxAvailableSizeWithAlignment (nuint alignment);
```

Comment:
prefix with `Get` (no verb in method) and remove `WithAlignment` suffix (implied by parameter)